src/components/WalletModal/index.tsx

```diff
@@ -185,9 +128,9 @@ export default function WalletModal({ openSettings }: { openSettings: () => void
         <ThemedText.SubHeader fontWeight={500}>Connect a wallet</ThemedText.SubHeader>
         <IconButton Icon={Settings} onClick={openSettings} data-testid="wallet-settings" />
       </AutoRow>
-      {pendingError ? (
-        pendingConnection && (
-          <ConnectionErrorView openOptions={openOptions} retryActivation={() => tryActivation(pendingConnection)} />
+      {pending.error ? (
+        pending.error && (
+          <ConnectionErrorView openOptions={cancelActivation} retryActivation={() => activate(pending.error)} />
```

Comment:
I personally find it a bit confusing that `openOptions` here is passed `cancelActivation`. What does `cancelActivation` do that it opens options? Is there a better / more explicit way to express it? For example, clearing errors and then, opening the modal once again? I guess (but it's just my guess and I would like some help here) that canceling activation works here only because we have an assumption that modal is already open? 

Could this potentially break if modal wasn't open? For connections such as WalletConnect, we may hide the modal while connection is "pending" to avoid z-index issues. Could this break?

Reply:
Good point, renaming from `openOptions` to `close`. `cancelActivation` resets the pending connection and error, causing this component to render the connection options again rather than showing the error view